bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BridgeHandler.java

```diff
@@ -886,6 +887,28 @@ public UserDefinedState getUserStateInfo(String stateId)
      */
     public <T extends BoschSHCServiceState> @Nullable Response putState(String deviceId, String serviceName, T state)
             throws InterruptedException, TimeoutException, ExecutionException {
+        return sendState(deviceId, serviceName, state, PUT);
+    }
+
+    public <T extends BoschSHCServiceState> @Nullable Response postState(String deviceId, String serviceName, T state)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        return sendState(deviceId, serviceName, state, POST);
+    }
+
+    /**
+     * Sends a state change for a device to the controller
+     *
+     * @param deviceId Id of device to change state for
+     * @param serviceName Name of service of device to change state for
+     * @param state New state data to set for service
+     *
+     * @return Response of request
+     * @throws InterruptedException
+     * @throws ExecutionException
+     * @throws TimeoutException
+     */
+    public <T extends BoschSHCServiceState> @Nullable Response sendState(String deviceId, String serviceName, T state,
```

Comment:
Does it need to be public? It seems the only callers are the two public methods `putState` and `postState`, so I guess it could be private and you could provide Javadoc for those instead?

Reply:
You're right, I did not want to duplicate the code and refactored the method. While doing this I just left the original modifier. But `private` is possible and I therefore changed it.